frontend/packages/dev-console/src/components/catalog/hooks/useCatalogCategories.ts

```diff
@@ -1,6 +1,29 @@
+import * as React from 'react';
+
 import { defaultCatalogCategories } from '../utils/default-categories';
-import { CatalogCategories } from '../utils/types';
+import { CatalogCategory } from '../utils/types';
+
+const useCatalogCategories = (): CatalogCategory[] => {
+  const categories = React.useMemo<CatalogCategory[]>(() => {
+    try {
+      const categoriesString = window.SERVER_FLAGS.developerCatalogCategories;
+      if (!categoriesString) {
+        return defaultCatalogCategories;
+      }
+
+      const categoriesArray: CatalogCategory[] = JSON.parse(categoriesString);
+
+      if (!Array.isArray(categoriesArray)) {
+        return defaultCatalogCategories;
+      }
+
+      return categoriesArray;
+    } catch (error) {
+      return defaultCatalogCategories;
```

Comment:
Again we shouldn't swallow the error. That makes it impossible to debug problems. At the very least, this needs to be logged in the console.

Reply:
Add console.error logging